### PR TITLE
MINOR: move topic configuration defaults

### PR DESCRIPTION
**Pull request by cmccabe**
Add default values for the configurations to TopicConfig.java.  Eventually we want to move
all of the log configurations into TopicConfig.java, and this is a good step along the way.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 1]
```diff
@@ -166,19 +191,36 @@
...
         "does not apply to any message format conversion that might be required for replication to followers.";
+
+    public static final String LEADER_REPLICATION_THROTTLED_REPLICAS = "leader.replication.throttled.replicas";
+    public static final Collection<String> LEADER_REPLICATION_THROTTLED_REPLICAS_DEFAULT = Collections.emptyList();
+    public static final String LEADER_REPLICATION_THROTTLED_REPLICAS_DOC = "A list of replicas for which log " +
+        "replication should be throttled on the leader side. The list should describe a set of replicas in the form " +
+        "[PartitionId]:[BrokerId],[PartitionId]:[BrokerId]:... or alternatively the wildcard '*' can be used to throttle " +
+        "all replicas for this topic.";
+
+    public static final String FOLLOW_REPLICATION_THROTTLED_REPLICAS = "follower.replication.throttled.replicas";
```
Comment (kamalcph) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:220]: typo: FOLLOW_REPLICATION_THROTTLED_REPLICAS -> FOLLOWER_REPLICATION_THROTTLED_REPLICAS

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 2]
```diff
@@ -56,26 +66,30 @@
...
 
     public static final String RETENTION_MS_CONFIG = "retention.ms";
+    public static final long RETENTION_MS_DEFAULT = TimeUnit.DAYS.toMillis(7);
     public static final String RETENTION_MS_DOC = "This configuration controls the maximum time we will retain a " +
         "log before we will discard old log segments to free up space if we are using the " +
         "\"delete\" retention policy. This represents an SLA on how soon consumers must read " +
         "their data. If set to -1, no time limit is applied.";
 
     public static final String MAX_MESSAGE_BYTES_CONFIG = "max.message.bytes";
+    public static final int MAX_MESSAGE_BYTES_DEFAULT = 1024 * 1024 + 12;
```
Comment (kamalcph) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:92]: Why not referring to Records.LOG_OVERHEAD instead of 12?

**Review by kamalcph — APPROVED** [review]

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java (hunk 1 above)
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:220]: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/LogConfig.scala#L150 can reference to this new field.

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 3]
```diff
@@ -166,19 +191,36 @@
...
 
     public static final String MESSAGE_DOWNCONVERSION_ENABLE_CONFIG = "message.downconversion.enable";
+    public static final boolean MESSAGE_DOWNCONVERSION_ENABLE_DEFAULT = true;
     public static final String MESSAGE_DOWNCONVERSION_ENABLE_DOC = "This configuration controls whether " +
         "down-conversion of message formats is enabled to satisfy consume requests. When set to <code>false</code>, " +
         "broker will not perform down-conversion for consumers expecting an older message format. The broker responds " +
         "with <code>UNSUPPORTED_VERSION</code> error for consume requests from such older clients. This configuration" +
         "does not apply to any message format conversion that might be required for replication to followers.";
+
+    public static final String LEADER_REPLICATION_THROTTLED_REPLICAS = "leader.replication.throttled.replicas";
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:213]: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/LogConfig.scala#L149 can reference to this field

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 4]
```diff
@@ -125,18 +145,21 @@
     public static final String CLEANUP_POLICY_CONFIG = "cleanup.policy";
     public static final String CLEANUP_POLICY_COMPACT = "compact";
     public static final String CLEANUP_POLICY_DELETE = "delete";
+    public static final String CLEANUP_POLICY_DEFAULT = CLEANUP_POLICY_DELETE;
     public static final String CLEANUP_POLICY_DOC = "A string that is either \"" + CLEANUP_POLICY_DELETE +
         "\" or \"" + CLEANUP_POLICY_COMPACT + "\" or both. This string designates the retention policy to use on " +
         "old log segments. The default policy (\"delete\") will discard old segments when their retention " +
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:151]: How about replacing `\"delete\"` by `CLEANUP_POLICY_DEFAULT`?

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 5]
```diff
@@ -85,31 +99,37 @@
         "limit only applies to a single record in that case.";
 
     public static final String INDEX_INTERVAL_BYTES_CONFIG = "index.interval.bytes";
-    public static final String INDEX_INTERVAL_BYTES_DOCS = "This setting controls how frequently " +
+    public static final int INDEX_INTERVAL_BYTES_DEFAULT = 4096;
+    public static final String INDEX_INTERVAL_BYTES_DOC = "This setting controls how frequently " +
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:103]: I want to see this change gets merged since I love consistency. However, this change may break the compatibility as this field is public.

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java [hunk 6]
```diff
@@ -166,19 +191,36 @@
...
     public static final String MESSAGE_DOWNCONVERSION_ENABLE_CONFIG = "message.downconversion.enable";
+    public static final boolean MESSAGE_DOWNCONVERSION_ENABLE_DEFAULT = true;
     public static final String MESSAGE_DOWNCONVERSION_ENABLE_DOC = "This configuration controls whether " +
         "down-conversion of message formats is enabled to satisfy consume requests. When set to <code>false</code>, " +
         "broker will not perform down-conversion for consumers expecting an older message format. The broker responds " +
         "with <code>UNSUPPORTED_VERSION</code> error for consume requests from such older clients. This configuration" +
         "does not apply to any message format conversion that might be required for replication to followers.";
+
+    public static final String LEADER_REPLICATION_THROTTLED_REPLICAS = "leader.replication.throttled.replicas";
+    public static final Collection<String> LEADER_REPLICATION_THROTTLED_REPLICAS_DEFAULT = Collections.emptyList();
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java:214]: The origin type is `List` rather than `Collection`. Is this change intentional?

**Review by chia7712 — COMMENTED** [review]
@cmccabe thanks for nice refactor. a couple of comments below.
